metagraph/tests/algorithms/test_embedding.py

```diff
@@ -130,56 +170,363 @@ def cmp_func(embedding):
-#         cmp_func
-#     )
+    embedding = mv.compute(
+        "embedding.train.graph_sage.mean",
+        graph,
+        node_feature_matrix,
+        node_feature_nodes,
+        walk_length,
+        walks_per_node,
+        layer_sizes,
+        samples_per_layer,
+        epochs,
+        learning_rate,
+        batch_size,
+    ).normalize(dpr.types.GraphSageNodeEmbedding.StellarGraphGraphSageNodeEmbeddingType)
+
+    # Create Testing Graph
+    unseen_a_nodes = np.arange(8888_00, 8888_20)
+    unseen_b_nodes = np.arange(8888_20, 8888_50)
+    unseen_complete_graph_a = nx.complete_graph(unseen_a_nodes)
+    unseen_complete_graph_b = nx.complete_graph(unseen_b_nodes)
+    unseen_nx_graph = nx.compose(unseen_complete_graph_a, unseen_complete_graph_b)
+    unseen_nx_graph.add_edge(8888_00_20, 8888_00)
+    unseen_nx_graph.add_edge(8888_00_20, 8888_20)
+    unseen_node_feature_np_matrix = np.zeros([51, embedding_size])
+    unseen_node_feature_np_matrix[0:20, 0] = 1
+    unseen_node_feature_np_matrix[20:50, 1] = 1
+    unseen_node_feature_matrix = dpr.wrappers.Matrix.NumpyMatrix(
+        unseen_node_feature_np_matrix
+    )
+    unseen_node_feature_nodes = dpr.wrappers.NodeMap.NumpyNodeMap(
+        np.arange(51),
+        node_ids=np.concatenate(
+            [unseen_a_nodes, unseen_b_nodes, np.array([8888_00_20])]
+        ),
+    )
+    unseen_graph = dpr.wrappers.Graph.NetworkXGraph(unseen_nx_graph)
+    matrix = mv.transform(
+        dpr.plugins.metagraph_stellargraph.algos.util.graph_sage_node_embedding.apply,
+        embedding,
+        unseen_graph,
+        unseen_node_feature_matrix,
+        unseen_node_feature_nodes,
+        batch_size=batch_size,
+        worker_count=1,
+    )
+
+    # Verify GraphSAGE results
+    def cmp_func(matrix):
+        assert tuple(matrix.shape) == (51, layer_sizes.as_dense(copy=False)[-1])
+        np_matrix = matrix.as_dense(copy=False)
+        unseen_a_vectors = np_matrix[0:20]
+        unseen_b_vectors = np_matrix[20:50]
+
+        _, neighbor_indices = (
+            NearestNeighbors(n_neighbors=20).fit(np_matrix).kneighbors(np_matrix)
+        )
+
+        for unseen_a_node_position in range(20):
+            unseen_a_node_neighbor_indices = neighbor_indices[unseen_a_node_position]
+            for unseen_a_node_neighbor_index in unseen_a_node_neighbor_indices:
+                assert 0 <= unseen_a_node_neighbor_index < 20
+
+        for unseen_b_node_position in range(20, 50):
+            unseen_b_node_neighbor_indices = neighbor_indices[unseen_b_node_position]
+            for unseen_b_node_neighbor_index in unseen_b_node_neighbor_indices:
+                assert 20 <= unseen_b_node_neighbor_index < 50
+
+    matrix.normalize(dpr.types.Matrix.NumpyMatrixType).custom_compare(cmp_func)
+
+
+def test_line(default_plugin_resolver):
+    try:
+        from sklearn.mixture import GaussianMixture
+    except:
```

Comment:
`except ImportError:`

Reply:
https://github.com/metagraph-dev/metagraph/pull/138/commits/c6f88dfd8b708da632b8f4336400a510ed49f371 makes it so that it catches `ModuleNotFoundError`.